### ISSUE TITLE
Consider adding a repo-wide namespace.

### ISSUE THREAD

**Issue by kylesemelka-figure** [miker2/flexi_config_reader#81]
For example, `namespace flexi_cfg`. This would encapsulate the types instead of polluting the global namespace.

**Comment by miker2** [comment]
Good call. most things are in the `config` namespace, but it looks like a few things didn't end up there. I'll go through and fix that. Is `config` okay as the top level namespace, or would `flexi_cfg` be better?

**Comment by miker2** [comment]
Put up a PR for this. It would seem that a bit more rework might be required to clean up the namespaces a bit (perhaps matching namespaces with directories), but everything is now within the `flexi_cfg` namespace.
